crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs

```diff
@@ -93,12 +100,20 @@ pub mod internal {
 
     pub(super) fn setup_plugin(app: &mut App) {
         app.add_systems(Startup, setup_system)
-            .add_systems(First, launch_diagnostic_tasks)
-            .add_systems(Update, read_diagnostic_tasks)
-            .init_resource::<SysinfoTasks>();
+            .add_systems(First, wake_diagnostic_task)
+            .add_systems(Update, read_diagnostic_task);
     }
 
-    fn setup_system(mut diagnostics: ResMut<DiagnosticsStore>) {
+    fn setup_system(mut diagnostics: ResMut<DiagnosticsStore>, mut commands: Commands) {
+        let (tx, rx) = mpsc::channel();
+        let diagnostic_task = DiagnosticTask::new(tx);
+        let waker = Arc::clone(&diagnostic_task.waker);
+        AsyncComputeTaskPool::get().spawn(diagnostic_task).detach();
```

Comment:
It may be better to just embed the `Task<()>` handle into `SysInfoTask`, just to ensure that the task is terminated if/when the resource itself is dropped for whatever reason.